### PR TITLE
captureDiagnosticsForCancelledRntbdRequests

### PR DESCRIPTION
**Pull request by xinlian12**
**Changes:**
Capture the diagnostics for cancelled rntbd requests. 

**Before the change:**
image

**After the change:**
image

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentServiceRequestContext.java [hunk 1]
```diff
@@ -44,6 +45,9 @@ public class DocumentServiceRequestContext implements Cloneable {
     public volatile boolean replicaAddressValidationEnabled = Configs.isReplicaAddressValidationEnabled();
     private final Set<Uri> failedEndpoints = ConcurrentHashMap.newKeySet();
 
+    // For cancelled rntbd requests, track the response as OperationCancelledException which later will be used to populate the cosmosDiagnostics
+    public volatile Map<String, CosmosException> rntbdCancelledRequestMap = new ConcurrentHashMap<>();
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentServiceRequestContext.java:49]: why not final?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GoneException.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public GoneException(String message, URI requestUri, Exception cause, int subSta
     }
 
     /**
-     * Instantiates a new Gone exception.
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GoneException.java]: revert

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GoneException.java]: fixed

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12

**Comment by FabianMeiswinkel** [comment]
CI failures due to known issue with CI pipeline

**Pull request merged**